### PR TITLE
Jim Macur

### PR DESCRIPTION
**Pull request by jimmacur**
Here is my IC 4. I was close to completing the 3rd iteration but did not get a chance to make it pass the last test.

### CODE REVIEW

**Comment by jimmacur** [comment]
Also, I had a break in time to take care of some life stuff, which is why my time may seem long. But it should show that my total time was around 3 hours.

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        @exhibits.each do |exhibit|
+            patrons_by_exhibit_interest[exhibit] = interested_patrons(exhibit)
+        end
+        patrons_by_exhibit_interest
+    end
+
+    def interested_patrons(exhibit)
+        @patrons.select do |patron|
+            patron.interests.include?(exhibit.name)
+        end
```
[the remark this answers is not in the document]
Comment (KatBrandt) [reply lib/museum.rb:40]: Make sure you write unit tests for your helper methods too though, I don't see a test for this method.

File: lib/museum.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
+class Museum
+    attr_reader :name, :exhibits, :patrons
+
+    def initialize(name)
+        @name = name
+        @exhibits = []
+        @patrons = []
+        @ticket_lottery_contestants = []
```
[the remark this answers is not in the document]
Comment (jimmacur) [reply lib/museum.rb:8]: That makes sense. I think I made it at the beginning with something in mind but didn't delete it after I didn't use it.

File: lib/museum.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jimmacur) [reply lib/museum.rb:40]: That makes sense. I think I was so focused on the method it was helping that I didn't consider writing a test for that helper method.

File: spec/museum_spec.rb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        it 'can list patrons by exhibit interest' do
+            @dmns.add_exhibit(@gems_and_minerals)
+            @dmns.add_exhibit(@dead_sea_scrolls)
+            @dmns.add_exhibit(@imax)
+
+            @dmns.admit(@patron_1)
+            @dmns.admit(@patron_2)
+            @dmns.admit(@patron_3)
+
+            expect(@dmns.patrons_by_exhibit_interest).to be_a Hash
```
[the remark this answers is not in the document]
Comment (jimmacur) [reply spec/museum_spec.rb:83]: Thank you. I refactored that test to make it actually reflect the hash that is there.

File: spec/museum_spec.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        end
+
+        xit 'has an empty array for lottery ticket contestants' do
```
[the remark this answers is not in the document]
Comment (jimmacur) [reply spec/museum_spec.rb:86]: I am not exactly sure, I have it un-skipped now.

**Pull request closed without merging**